### PR TITLE
[Skeleton Tool] Hide "Make Parent To Active Bone" menu item when not using Skeleton Tool

### LINKED ISSUE #1669: [Skeleton Tool] Hide "Make Parent To Active Bone" menu item when not using Skeleton Tool
The "Make Parent To Active Bone" menu item appears when user right-clicks on any control point of bone.
This menu item should be visible only when Skeleton Tool is active.

### PR DESCRIPTION
**Pull request by Aa20475**
Closes #1669

### CODE REVIEW

**Comment by morevnaproject** [comment]
I wonder, is there any better way to identify current active tool, instead of adding `get_canvas_interface()->set_state()` method?

@rodolforg @ice0 any advice here? ^__^

**Comment by ice0** [comment]
@morevnaproject you can use something like:
```
canvas_view->get_smach().get_state_name() == "bline"
```

**Comment by Aa20475** [comment]
But we can't access `canvas_view` in actions right?

**Comment by ice0** [comment]
I think you can use `App::get_selected_canvas_view()` for that.

**Comment by morevnaproject** [comment]
> I think you can use `App::get_selected_canvas_view()` for that.
> 
> 
> 
> Also I think maybe add a method to Toolbox to get it state, as it looks more intuitive for me. Something like:
> 
> 
> 
> ```
> 
> App::get_toolbox()->get_state_name(); // returns current state name
> 
> ```
> 
> 
> 
> What do you think guys?
> 
> 

Maybe even simpler? ^__^ -
'''
App::get_state_name()
'''

**Comment by morevnaproject** [comment]
> I think you can use `App::get_selected_canvas_view()` for that.

@ice0 Unfortunately, we can't use `App::` in actions (as they are part of `synfigapp`, not `gui`).

**Comment by ice0** [comment]
@morevnaproject `CanvasInterface` is a part of `synfigapp` so anyway you need it.

**Comment by morevnaproject** [comment]
@ice0 So, do you suggest to keep the current solution as is?

**Comment by ice0** [comment]
The only improvement I can suggest is to remove `CanvasInterface::set_state` and `CanvasInterface::get_state` and use `smach` for that.

**Comment by morevnaproject** [comment]
I agree with that, but how to access `smach` data from action of `synfigapp`?

**Comment by rodolforg** [comment]
Hi.
AFAIK `synfigapp` is focused on change the "synfig document" behind the curtains– I mean, the GUI ;)
Change the current tool is not such an action, as well as "select specific layer" is not too.

So _maybe_ `synfigapp` is not the right place?

**Comment by ice0** [comment]
@rodolforg @morevnaproject I think you are right guys. So I need a time to think about it. Maybe someone will suggest a better solution.

**Comment by rodolforg** [comment]
Well, PR #1698 was merged and it has this same `CanvasInterface::set_state` and `CanvasInterface::get_state` on master branch...
So I think we can merge this PR and decide about it later .

File: synfig-studio/src/synfigapp/actions/valuedescbonesetparent.cpp [hunk 1]
```diff
@@ -103,10 +103,14 @@ Action::ValueDescBoneSetParent::is_candidate(const ParamList &x)
...
 
+	i = x.find("canvas_interface");
+	etl::handle<CanvasInterface> ci = i->second.get_canvas_interface();
+
 	return value_desc.parent_is_value_node()
 		&& ValueNode_Bone::Handle::cast_dynamic(value_desc.get_parent_value_node())
 		&& child
-		&& ValueNode_Bone::Handle::cast_dynamic(child);
+		&& ValueNode_Bone::Handle::cast_dynamic(child)
+		&& ci->get_state()=="bone";
```
Comment (rodolforg) [inline synfig-studio/src/synfigapp/actions/valuedescbonesetparent.cpp:113]: maybe check `&& ci`  before `&& ci->get_state()=="bone";`

**Comment by rodolforg** [comment]
Superseded by #2064

**Pull request closed without merging**